### ISSUE TITLE
tcltk example crash

### ISSUE THREAD

**Issue by ovenpasta** [feeley/gambit#164]
I compiled the tcltk example provided with the distribution. 
I'm using Linux Mint with x86_64 architecture
I had to modify the makefile to properly link with tcl8.5 and tk8.5. Version 8.6 is not working. I get some error:
```
tcltk.c: In function ‘tcltk_eval’:
tcltk.c:1133:37: error: ‘Tcl_Interp’ has no member named ‘result’
         tcltk_eval_error (tcl_interp->result);
```
With tcl8.5 builds properly but at the end of the build I get this:
```
../../gsi/gsi -:~~bin=../../bin,~~lib=../../lib,~~include=../../include tcltk -e '(load "test") (main)'
Segmentation fault (core dumped)
make: *** [run] Errore 139
```

Then if I launch the tcltk example with valgrind I get this error:
```
$ valgrind gsi tcltk
==18454== Memcheck, a memory error detector
==18454== Copyright (C) 2002-2013, and GNU GPL'd, by Julian Seward et al.
==18454== Using Valgrind-3.10.0.SVN and LibVEX; rerun with -h for copyright info
==18454== Command: gsi tcltk
==18454== 
==18454== Invalid read of size 8
==18454==    at 0x4A187B: ___alloc_scmobj (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x4A1A32: ___make_vector (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x611844E: ___H__20_tcltk (in /home/aldo/scheme/gambit/gambit/examples/tcltk/tcltk.o1)
==18454==    by 0x49DAC3: ___call (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x49DC95: ___setup_pstate (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x49DCD4: ___setup_vmstate (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x49E12B: ___setup (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x4B417F: ___main (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x4A8AAC: ___main_char (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x49C80A: main (in /home/aldo/gambit/bin/gsi)
==18454==  Address 0x181 is not stack'd, malloc'd or (recently) free'd
==18454== 
==18454== 
==18454== Process terminating with default action of signal 11 (SIGSEGV)
==18454==  Access not within mapped region at address 0x181
==18454==    at 0x4A187B: ___alloc_scmobj (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x4A1A32: ___make_vector (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x611844E: ___H__20_tcltk (in /home/aldo/scheme/gambit/gambit/examples/tcltk/tcltk.o1)
==18454==    by 0x49DAC3: ___call (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x49DC95: ___setup_pstate (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x49DCD4: ___setup_vmstate (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x49E12B: ___setup (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x4B417F: ___main (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x4A8AAC: ___main_char (in /home/aldo/gambit/bin/gsi)
==18454==    by 0x49C80A: main (in /home/aldo/gambit/bin/gsi)
==18454==  If you believe this happened as a result of a stack
==18454==  overflow in your program's main thread (unlikely but
==18454==  possible), you can try to increase the size of the
==18454==  main thread stack using the --main-stacksize= flag.
==18454==  The main thread stack size used in this run was 8388608.
==18454== 
==18454== HEAP SUMMARY:
==18454==     in use at exit: 9,572,854 bytes in 1,456 blocks
==18454==   total heap usage: 2,244 allocs, 788 frees, 10,413,557 bytes allocated
==18454== 
==18454== LEAK SUMMARY:
==18454==    definitely lost: 0 bytes in 0 blocks
==18454==    indirectly lost: 0 bytes in 0 blocks
==18454==      possibly lost: 9,333,564 bytes in 323 blocks
==18454==    still reachable: 239,290 bytes in 1,133 blocks
==18454==         suppressed: 0 bytes in 0 blocks
==18454== Rerun with --leak-check=full to see details of leaked memory
==18454== 
==18454== For counts of detected and suppressed errors, rerun with: -v
==18454== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
Ucciso
```

**Comment by feeley** [comment]
I have fixed a bug in the tcltk examples (commit c2559c3a03be1eb611ba92415fe354dbcb85ad5b).  Can you check if it fixes your problem?

I have also committed a fix for tcltk 8.6 (commit 9e85e3b4ced2c356599ff8f704d3d9abf01d13f2).

**Comment by ovenpasta** [comment]
All fixed ;) thank you
